search/query.go

```diff
@@ -44,60 +47,150 @@ type PostSearchResult struct {
 	Post any        `json:"post"`
 }
 
-func doSearchPosts(ctx context.Context, escli *es.Client, q string, offset int, size int) (*EsSearchResponse, error) {
+func checkParams(offset, size int) error {
+	if offset+size > 5000 || size > 1000 || offset > 1000 || offset < 0 || size < 0 {
```

Comment:
Are these a bit redundant? `offset+size` can't be > `2000` with the explicit size and offset rules, right?

Reply:
yeah, it is redundant. I guess should decide whether we want this check here or in the other place. I like having it just before query to have confidence it is always applied; but also verifying when HTTP requests are parsed is idiomatic. hrm.

Reply:
The check is fine you're just checking x+y > 5000 || x > 1000 || y > 1000 which is redundant since x+y will only be > 5000 if one of them is > 1000.

Reply:
oh, hah, yeah. updated with better and no-longer-redundant sanity check:

`offset+size > 10000 || size > 250 || offset > 10000 || offset < 0 || size < 0`

Reply:
(well, I guess it is still slightly redundant, but :shrug:)